Mathlib/Algebra/CharP/Two.lean

```diff
@@ -20,15 +20,34 @@ variable {R ι : Type*}
 
 namespace CharTwo
 
+section AddMonoidWithOne
+
+variable [AddMonoidWithOne R]
+
+theorem two_eq_zero [CharP R 2] : (2 : R) = 0 := by
+  rw [← Nat.cast_two, CharP.cast_eq_zero]
+
+/-- The only hypotheses required to build a `CharP R 2` instance are `1 ≠ 0` and `2 = 0`. -/
+protected theorem mk (h₁ : (1 : R) ≠ 0) (h₂ : (2 : R) = 0) : CharP R 2 := by
```

Comment:
I'm not sure about this name, `mk` is usually the constructor of a structure.

Reply:
This is meant as a pseudo-constructor of sorts. I think the name is fine, since I don't think there's going to be any alternate ways of building the instance that don't just boil down to this, and alternate names like `char_two_of_one_ne_zero_of_two_eq_zero` are way too wordy.